### ISSUE TITLE
Swagger module with RestEasy

### ISSUE THREAD

**Issue by JLLeitschuh** [gwizard/gwizard#18]
I'm starting a new project using REST, I initially tried to use jersey 2 and ran into the massive conundrum that was guice with hk2. So I decided, after looking at your project, to use RESTeasy instead.

I was able to get jersey 2 to work before I threw it all away to use RESTeasy instead.

If I'm using RESTeasy with swagger do I need the RESTeasy module installed? Do I just use the Swagger module? Is there some subtle thing I need to do?

According to your [readme for the swagger module](https://github.com/gwizard/gwizard/blob/master/gwizard-swagger/README.md) the first argument of your main method needs to be your swagger.yml file? Is there some subtle reflection magic that I don't know about going on here to make the input arguments to your program end up somewhere else?

I'm trying to launch this whole thing with Jetty using gradle. The project is assembled into a WAR and the `GuiceServletContextListener` module is initialized via the web.xml file.

Long story short, something isn't working here and I'm not sure what.

**Comment by JLLeitschuh** [comment]
I solved this problem by copying and pasting the contents of the `RestModule` with my own modifications.

```
private static Module createRestModule() {
  return new ServletModule() {

    @Override
    protected void configureServlets() {
      install(new JaxrsModule());

      // Binding this will cause it to be picked up by gwizard-web
      bind(GuiceResteasyBootstrapServletContextListener.class);

      // Make sure RESTEasy picks this up so we get our ObjectMapper from guice
      bind(ObjectMapperContextResolver.class);

      bind(HttpServletDispatcher.class)
          .in(Singleton.class);
      serve("/*")
          .with(HttpServletDispatcher.class);
    }
  };
}
```

This is then installed in the `GuiceServletContextListener`.

In the `getInjector()` method of the `GuiceServletContextListener` I use the `EventListenerScanner` like this:

```java
eventListenerScanner
        = injector.getInstance(EventListenerScanner.class);
```

My `contextInitialized` and `contextDestroyed` respectively each look like this:
```java
@Override
public void contextInitialized(ServletContextEvent servletContextEvent) {
  // Assign this first. getInjector is called in the super method.
  super.contextInitialized(servletContextEvent);
  // This will be initialized by now
  eventListenerScanner.accept(visit -> {
    if (visit instanceof ServletContextListener) {
      ((ServletContextListener) visit).contextInitialized(servletContextEvent);
    }
  });
}

@Override
public void contextDestroyed(ServletContextEvent servletContextEvent) {
  if (eventListenerScanner != null) {
    eventListenerScanner.accept(visit -> {
      if (visit instanceof ServletContextListener) {
        ((ServletContextListener) visit).contextInitialized(servletContextEvent);
      }
    });
  } else {
    LOGGER.error("eventListenerScanner was null. Injector must have failed to be created");
  }
  super.contextDestroyed(servletContextEvent);
}
```

This allows whatever servlet container you are using to construct your injector. Then you can use `Guice` to instantiate the rest of the modules that need to be instantiated.
Then you simply pass the `contextInitialized` and `contextDestroyed` messages to the listeners that are now under your control.

There may be a better way to do this with a seperate module, or by creating a `GwizardGuiceServletContextListener` but this works for me for now.

**Comment by stickfigure** [comment]
Closing out old tickets - the swagger module was contributed; it has been removed for 1.0 pending someone that wants to maintain it.
